docs/how-to/package-with-hatch.md

```diff
@@ -0,0 +1,380 @@
+
+Bear this in mind, if your project name contains spaces.
+
+
+### Got existing code?
+
+The `hatch new` command assumes you are starting from scratch.
+If you have existing code you would like to package, navigate to your directory and run the following command:
+
+```
+hatch new --init <YOUR PROJECT NAME>
+```
+
+This will generate a `pyproject.toml` in the current directory.
+
+!!! note
+    
+    It will simplify things if your code follows the directory structure convention above. This may require that you move your files -- you only need to do this once!
+
+## Adding code
+
+Your code should reside inside `src/<PROJECT NAME>`.
+For the calculator example we will copy `calculator.py` and `calculator.tcss` into the `src/textual_calculator` directory, so our directory will look like the following:
+
+```
+textual-calculator
+├── LICENSE.txt
+├── README.md
+├── pyproject.toml
+├── src
+│   └── textual_calculator
+│       ├── __about__.py
+│       ├── __init__.py
+│       ├── calculator.py
+│       └── calculator.tcss
+└── tests
+    └── __init__.py
+```
+
+## Adding dependencies
+
+Your Textual app will likely depend on other Python libraries (at the very least Textual itself).
+We need to list these in `pyproject.toml` to ensure that these *dependencies* are installed alongside your app.
+
+In `pyproject.toml` there should be a section beginning with `[project]`, which will look something like the following:
+
+```toml
+[project]
+name = "textual-calculator"
+dynamic = ["version"]
+description = 'A example app'
+readme = "README.md"
+requires-python = ">=3.8"
+license = "MIT"
+keywords = []
+authors = [
+  { name = "Will McGugan", email = "redacted@textualize.io" },
+]
+classifiers = [
+  "Development Status :: 4 - Beta",
+  "Programming Language :: Python",
+  "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: 3.12",
+  "Programming Language :: Python :: Implementation :: CPython",
+  "Programming Language :: Python :: Implementation :: PyPy",
+]
+dependencies = []
+```
+
+We are interested in the `dependencies` value, which should list the app's dependencies.
+If you want a particular version of a project you can add `==` followed by the version.
+
+For the calculator, the only dependency is Textual.
+We can add Textual by modifying the following line:
+
+```toml
+dependencies = ["textual==0.47.1"]
+```
+
+At the time of writing, the latest Textual is `0.47.1`.
+The entry in `dependencies` will ensure we get that particular version, even when newer versions are released.
+
+See the Hatch docs for more information on specifying [dependencies](https://hatch.pypa.io/latest/config/dependency/).
```

Comment:
Shouldn't we encourage reading how Hatch tells you to specify dependencies more strongly and earlier than we mention that you can edit that list by hand and type`==` to pin versions?

Reply:
This sentence is so close to the others, that I think it will be fresh in the readers mind.